internal/controller/streams.go

```diff
@@ -328,25 +328,49 @@ func (sm *streamManager) StreamResources(ctx context.Context, logger *zap.Logger
 	}
 	logger.Info("Successfully sent resource snapshot")
 
+	mutationChan := make(chan *pb.KubernetesResourceMutation)
```

Comment:
This channel is leaked?
```suggestion
	mutationChan := make(chan *pb.KubernetesResourceMutation)

	defer close(mutationChan)
```

Hmmm this is tricky since you must wait until all the writers are done, and can't write into the channel anymore, before closing the channel.
What would be a good place to close it?
You probably need to use a `WaitGroup` to wait for all the `WatchK8sResources` goroutines to return?